dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -126,13 +137,38 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::Unregiste
     auto lock = m_lock.lock_exclusive();
     THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
 
-    const std::wstring appId = GetAppIdentifier(processName);
+    const std::wstring appId{ GetAppIdentifier(processName) };
     m_foregroundSinkManager->Remove(appId);
 
     return S_OK;
 }
 CATCH_RETURN()
 
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::AddToastRegistrationMapping(_In_ PCWSTR processName, _In_ PCWSTR toastAppId) noexcept try
+{
+    auto lock{ m_lock.lock_exclusive() };
+    THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
+
+    // Don't register unless there is a sink available.
+    if (GetAppIdentifier(processName).empty())
+    {
+        return S_OK;
+    }
+    m_toastRegistrationManager->Add(processName, toastAppId);
```

Comment:
If this returns HRESULT, we should THROW_IF_FAILED

Reply:
ToastRegistrationManager.Add is void function.